references/detection/transforms.py

```diff
@@ -401,3 +401,39 @@ def forward(self, img, target=None):
             img, target = self._pad(img, target, [0, 0, pad_right, pad_bottom])
 
         return img, target
+
+
+class RandomShortestSize(nn.Module):
+    def __init__(
+        self,
+        min_size: Union[List[int], Tuple[int], int],
+        max_size: int,
+        interpolation: InterpolationMode = InterpolationMode.BILINEAR,
+    ):
+        super().__init__()
+        self.min_size = [min_size] if isinstance(min_size, int) else list(min_size)
+        self.max_size = max_size
+        self.interpolation = interpolation
+
+    def forward(
+        self, image: Tensor, target: Optional[Dict[str, Tensor]] = None
+    ) -> Tuple[Tensor, Optional[Dict[str, Tensor]]]:
+        _, orig_height, orig_width = F.get_dimensions(image)
+
+        min_size = self.min_size[torch.randint(len(self.min_size), (1,)).item()]
+        r = min(min_size / min(orig_height, orig_width), self.max_size / max(orig_height, orig_width))
+
+        new_width = int(orig_width * r)
+        new_height = int(orig_height * r)
```

Comment:
D2 also does +0.5 here: 
https://github.com/facebookresearch/detectron2/blob/9258799e4e72786edd67940872e0ed2c4387aac5/detectron2/data/transforms/augmentation_impl.py#L193-L194

Reply:
Thanks for the comment. You are right to say that D2 is doing round in this way (it's a common idiom in their codebase) but here I'm porting TorchVision's [GeneralizedRCNNTransform.resize()](https://github.com/pytorch/vision/blob/main/torchvision/models/detection/transform.py#L163) which works very similarly but is not identical. My target is to export this transform from within the model and move it outside so that we can use other transformations such as LSJ.